### PR TITLE
add(functions): add how-to retention for function triggers SRVLESS-9134

### PR DESCRIPTION
**Pull request by miguescri**
### Your checklist for this pull request

- [ x] Check that the commit messages match our requested structure.
- [ x] Name your pull request according to the [contribution guidelines](https://github.com/scaleway/docs-content/blob/main/docs/CONTRIBUTING.md).

### Description
Please describe your pull request.

### CODE REVIEW

File: serverless/functions/how-to/add-trigger-to-a-function.mdx [hunk 1]
```diff
@@ -41,6 +41,8 @@ For each message that is sent to an SQS queue, the SQS trigger reads the message
 The function can then process the message and perform any required actions, such as updating a database or sending a notification.
 Currently, SQS queues of the [Scaleway Messaging and Queuing](https://www.scaleway.com/en/docs/serverless/messaging/quickstart/) product are supported.
 
+The configuration of the queue retention may affect the behavior of the trigger. See [How to configure event retention for trigger inputs](/serverless/functions/reference-content/configure-trigger-inputs/).
```
Comment (SamyOubouaziz) [inline serverless/functions/how-to/add-trigger-to-a-function.mdx:44]: Refer to the [How to configure event retention for trigger inputs](/serverless/functions/reference-content/configure-trigger-inputs/) page for more information.

File: serverless/functions/reference-content/configure-trigger-inputs.mdx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+categories:
+  - serverless
+  - functions
+---
+
+<Message type="important">
+  This feature is currently in beta and only allows a max of 10 inflight requests, so functions that are configured to have more than 10 replicas won't be used at full. In the future this limit will be removed. Please, take that into consideration while using the following formulas.
+</Message>
+
+Triggers get events from an input, such as a SQS queue, and forward them to a function, which will scale up according to its settings to accommodate the workload. This process uses back pressure, so that no new events are read if the function instances have not yet finished processing the previous ones. It is important to note that triggers only keep a buffer of the messages that are inflight, and they DO NOT drain all the messages of the input in advance.
```
Comment (SamyOubouaziz) [inline serverless/functions/reference-content/configure-trigger-inputs.mdx:21]: Replace last sentence by the following
```
<Message type="note">
  triggers only keep a buffer of the messages that are inflight, they do not drain all the messages of the input in advance.
</Message>
```

File: serverless/functions/reference-content/configure-trigger-inputs.mdx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+As a result, in some scenarios such as event bursts or slow computations, events may stay in the input buffer for a while before being consumed. If there is some kind of message-level timeout in the input, as in the case of [MessageRetentionPeriod](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SetQueueAttributes.html) in SQS queues, this may lead to events being deleted before triggering the function.
+
+The implementation of the core trigger behavior strives to be input agnostic, so it is the responsibility of users to configure the input buffers according to their use case to avoid losing events. Also, some users may be okay with old messages being automatically deleted if they haven't been processed in time.
```
Comment (SamyOubouaziz) [inline serverless/functions/reference-content/configure-trigger-inputs.mdx:25]: "... core trigger behavior is input agnostic, it is therefore your responsibility to configure the input buffers according to your use case..."

File: serverless/functions/reference-content/configure-trigger-inputs.mdx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+dates:
+  validation: 2023-06-09
+  posted: 2023-06-09
+categories:
+  - serverless
+  - functions
+---
+
+<Message type="important">
+  This feature is currently in beta and only allows a max of 10 inflight requests, so functions that are configured to have more than 10 replicas won't be used at full. In the future this limit will be removed. Please, take that into consideration while using the following formulas.
```
Comment (SamyOubouaziz) [inline serverless/functions/reference-content/configure-trigger-inputs.mdx:18]: max > maximum

File: serverless/functions/reference-content/configure-trigger-inputs.mdx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+</Message>
+
+Triggers get events from an input, such as a SQS queue, and forward them to a function, which will scale up according to its settings to accommodate the workload. This process uses back pressure, so that no new events are read if the function instances have not yet finished processing the previous ones. It is important to note that triggers only keep a buffer of the messages that are inflight, and they DO NOT drain all the messages of the input in advance.
+
+As a result, in some scenarios such as event bursts or slow computations, events may stay in the input buffer for a while before being consumed. If there is some kind of message-level timeout in the input, as in the case of [MessageRetentionPeriod](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SetQueueAttributes.html) in SQS queues, this may lead to events being deleted before triggering the function.
```
Comment (SamyOubouaziz) [inline serverless/functions/reference-content/configure-trigger-inputs.mdx:23]: If the input messages in SQS queues are set with a timeout such as the [MessageRetentionPeriod](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SetQueueAttributes.html), events may be deleted before triggering the function.

Comment (miguescri) [reply serverless/functions/reference-content/configure-trigger-inputs.mdx:23]: Right now we only support SQS, but in the future we'll add more inputs and this issue affects them all, each with their own configuration set. This is just an example.

**Review by bene2k1 — APPROVED** [review]

**Pull request merged**